api/v1alpha1/capiprovider_types.go

```diff
@@ -79,13 +79,18 @@ type Features struct {
 // Credentials defines the external credentials information for the provider.
 // +kubebuilder:validation:MaxProperties=1
 // +kubebuilder:validation:MinProperties=1
+// +kubebuilder:validation:XValidation:message="rancherCloudCredentialNamespaceName should be in the namespace:name format.",rule="!has(self.rancherCloudCredentialNamespaceName) || self.rancherCloudCredentialNamespaceName.matches('^.+:.+$')"
 // +structType=atomic
 //
 //nolint:godot
+//nolint:lll
 type Credentials struct {
 	// RancherCloudCredential is the Rancher Cloud Credential name
 	RancherCloudCredential string `json:"rancherCloudCredential,omitempty"`
 
+	// RancherCloudCredentialNamespaceName is the Rancher Cloud Credential namespace:name reference
+	RancherCloudCredentialNamespaceName string `json:"rancherCloudCredentialNamespaceName,omitempty"`
```

Comment:
Maybe in the future with an api bump we can move both these under a `Rancher` struct

Reply:
Yeah, makes sense as a separate the field entirely.